### PR TITLE
Feature/aip 5624

### PR DESCRIPTION
**Pull request by fortunato**
Soft-delete mutations for elements/units/manifestations

### CODE REVIEW

File: src/schema/decomposition/element.repository.ts [hunk 1]
```diff
@@ -98,4 +98,15 @@ export class ElementRepository implements IElementRepository {
 			data,
 		});
 	}
+
+	async deleteElement(identifier: string): Promise<Element> {
+		const data: Prisma.elementsUpdateInput = {
+			deleted_at: new Date(),
```
Comment (dlnr) [inline src/schema/decomposition/element.repository.ts:104]: Not assignable to type elementsUpdateInput

File: src/schema/decomposition/manifestation.factory.ts [hunk 2]
```diff
@@ -33,6 +34,7 @@ export class ManifestationFactory {
 		manifestation.quantity = quantity;
 		manifestation.quantityUnitOfMeasurement = quantityUnitOfMeasurement;
 		manifestation.constructionYear = constructionYear;
+		manifestation.deletedAt = deletedAt instanceof Date ? deletedAt.toUTCString() : null;
```
Comment (dlnr) [inline src/schema/decomposition/manifestation.factory.ts:37]: Perhaps you can explain this error to me?

File: src/schema/decomposition/models/element.model.ts [hunk 3]
```diff
@@ -58,4 +58,7 @@ export class Element {
 
 	@Field((type) => Boolean)
 	isRelevant: boolean;
+
+	@Field((type) => String, { nullable: true })
+	deletedAt: string;
```
Comment (dlnr) [inline src/schema/decomposition/models/element.model.ts:63]: Should this be type Date everywhere?

**Review by dlnr — COMMENTED** [review]
I don't really get why you would assign the type never and the call it as a boolean or a Date. Perhaps you could enlighten me a little?

File: src/schema/decomposition/element.repository.spec.ts [hunk 4]
```diff
@@ -78,4 +78,17 @@ describe('ElementRepository', () => {
...
+		// @ts-ignore
+		prismaServiceMock.elements.update.mockResolvedValue(deletedElement);
+		const identifier = '610d0b4e-c06f-4894-9f60-8e1d0f78d2f1';
+		const repo = new ElementRepository(prismaServiceMock);
+		const element = await repo.deleteElement(identifier);
+		expect(prismaServiceMock.elements.update).toHaveBeenCalledWith({
+			where: { id: identifier },
+			data: expect.objectContaining({}),
+		});
+		expect(element.deleted_at instanceof Date).toBe(true);
```
[the remark this answers is not in the document]
Comment (fortunato) [reply src/schema/decomposition/element.repository.spec.ts:92]: I'd beg to differ. The test being green supports that notion. Why do you say that?

File: src/schema/decomposition/models/element.model.ts (hunk 3 above)
Comment (fortunato) [reply src/schema/decomposition/models/element.model.ts:63]: No. The GraphQL service can't communicate a date object in a query/mutation response. It has to be converted to a string for that reason. The client can turn it into a Date again though.

Comment (fortunato) [reply src/schema/decomposition/models/element.model.ts:63]: (this is a model that describes the GraphQL schema/interface)

**Review by anaim — APPROVED** [review]

**Pull request merged**